src/parse.c

```diff
@@ -23,27 +23,37 @@
 #include "qpid/dispatch/amqp.h"
 #include "qpid/dispatch/ctools.h"
 
+#include "buffer_field_api.h"
+
 #include <assert.h>
 #include <inttypes.h>
 #include <stdio.h>
 
 DEQ_DECLARE(qd_parsed_field_t, qd_parsed_field_list_t);
 
+
+typedef struct qd_amqp_field_t {
```

Comment:
I am thinking aloud here but can qd_amqp_field_t and qd_buffer_field_t just be combined into just a qd_amqp_field_t ?